sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/model/sql/adapter/BeamRecordSqlTypeBuilder.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.nexmark.model.sql.adapter;
+
+import com.google.common.collect.ImmutableList;
+
+import java.sql.Types;
+
+import org.apache.beam.sdk.extensions.sql.BeamRecordSqlType;
+
+/**
+ * Helper class to construct {@link BeamRecordSqlType}.
+ */
+public class BeamRecordSqlTypeBuilder {
```

Comment:
<!--thread_id:cc_150889287_t; commit:050bdab2afe544b8a5c58bfe1dfe1914a9820c8f; resolved:0-->
<!--section:context-quote-->
> **echauchot** wrote:
> This class seems to have nothing related to nexmark it should be in beam.sdk.extensions.sql package I think. But I'm not very sure this class is needed as it avoids only a few lines of code. WDYT?

<!--section:body-->
I wasn't sure whether it makes sense to add it to the main SQL package at this moment, don't know how useful it is for wider applications. I see no problems moving it though, if you think it belongs there.

Main reason I made it is for readability. Currently `BeamRecordSqlType.create()` only accepts separate lists of names and types, and it's hard to match them between each other, [like here](https://github.com/apache/beam/blob/39e66e953b0f8e16435acb038cad364acf2b3a57/sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/integrationtest/BeamSqlBuiltinFunctionsIntegrationTestBase.java#L65-L71). This class makes it clear which field has which type.

Reply:
Good point on the matching of the arguments. I think this kind of readability improvement could profit all the users of SQL DSL, might me worth a separate (because unrelated to nexmark) PR  toward SQL DSL. Have you considered putting the builder directly in the `BeamRecordSqlType` a bit like the `Autovalue` pattern for example?

Reply:
Sure, I'll move it to the main SQL package in a separate PR

Reply:
Implemented the builder in the main SQL package: https://github.com/apache/beam/pull/4168